src/test/ui/liveness/liveness-issue-77915.rs

```diff
@@ -0,0 +1,36 @@
+// Ensure inout asm! operands are marked as used by the liveness pass
+
+// only-x86_64
+// check-pass
+
+#![feature(asm)]
+#![allow(dead_code)]
+#![deny(unused_variables)]
+
+// Tests the single variable inout case
+unsafe fn rep_movsb(mut dest: *mut u8, mut src: *const u8, mut n: usize) -> *mut u8 {
```

Comment:
Can you also add a test that ensures projections work correctly as well?

Reply:
Added a couple of test cases for field projection, thanks!